### PR TITLE
Migrate flow around sphere example

### PR DESCRIPTION
**Pull request by CaroleAnneDaunais**
# Description of the problem

The flow around a sphere exmaple still had to be merged.

### CODE REVIEW

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+The following schematic describes the simulation and its boundary condition indices.
+
+.. image:: images/example_4_setup.png
+    :alt: The geometry and boundary conditions
+    :align: center
+    :name: geometry
+
+Note that the sphere surface has a boundary index of ``0``, the inlet ``1`` and the walls parallel to the flow direction have a boundary index of ``2``. 
+
+As this examples allows for two different Reynolds number as well as an adaptative mesh refinement variation, the Parameter file section will specify the differences when applicable.
```
Comment (tonielgeitani) [inline doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:34]:
```suggestion
As this examples allows for two different Reynolds numbers as well as an adaptative mesh refinement variation, the Parameter file section will specify the differences when applicable. 
```

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+Note that the sphere surface has a boundary index of ``0``, the inlet ``1`` and the walls parallel to the flow direction have a boundary index of ``2``. 
+
+As this examples allows for two different Reynolds number as well as an adaptative mesh refinement variation, the Parameter file section will specify the differences when applicable. 
+
+
+Parameter file
+--------------
+
+We first extablish the parameters.
```
Comment (tonielgeitani) [inline doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:40]:
```suggestion
We first establish the parameters.
```

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+
+.. warning:: 
+
+    If you choose to use the ``gd_navier_stokes_3d`` solver; only Q2-Q1 elements are supported. 
+
+
+Simulation control
+~~~~~~~~~~~~~~~~~~
+
+The parameters also slightly different from one problem to another differ in this section since as the different examples experience different flow regimes.
```
Comment (tonielgeitani) [inline doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:162]:
```suggestion
The parameters also slightly differ from one problem to another as different examples experience different flow regimes.
```

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+    set max cfl		              = 1000
+    set stop tolerance          = 1e-5
+    set adaptative time step scaling = 1.2
+    set number mesh adapt       = 0
+    set output name             = sphere-output
+    set output frequency        = 1
+    set subdivision             = 1
+  end
+
+The ``steady_bdf`` method solves for a steady-state simulation using adjoint time stepping with a bdf1 scheme. An initial time step is used to complete a transient iteration, and with each iteration, the time step is increased. The simulation is considered to have reached steady-state when the L2 norm of the initial residual is lower than stop tolerance at the start of a non-linear solution step, i.e. until the time step is large enough that a pseudo-steady-state has been reached.
```
Comment (tonielgeitani) [inline doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:193]:
```suggestion
The ``steady_bdf`` method solves for a steady-state simulation using adjoint time stepping with a bdf1 scheme. An initial time step is used to complete a transient iteration, and with each iteration, the time step is increased. The simulation is considered to have reached steady-state when the L2 norm of the initial residual is lower than the stop tolerance at the start of a non-linear solution step, i.e. until the time step is large enough that a pseudo-steady-state has been reached.
```

Comment (tonielgeitani) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:193]: You say two changes are implemented. You specify firstly, steady_bdf but you don't mention the second change.

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+    set fraction type	          = number
+    set max number elements     = 50000
+    set min refinement level    = 0
+    set max refinement level    = 3
+    set variable		            = velocity
+  end
+
+The mesh is dynamically adapted based on an estimate of the error of the solution for the velocity (the Kelly error estimator). The refinement is based on the number of elements. This means that the number of cells refined/coarsened per iteration is based on the fraction of the number of cells, rather than the fraction of the error (where all cells which have the fraction of the error are refined/coarsened).
+
+The ``min refinement level`` refers to the base mesh which has been used in the previous static simulations. The mesh can only become finer than this, not coarser. The ``max refinement level`` is set at 3, giving a maximum possible number of cells at 3 million. However, the ``max number elements`` limits the number of cells to 50,000 to keep the simulation within feasible computational expense.
```
Comment (tonielgeitani) [inline doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:229]:
```suggestion
The ``min refinement level`` refers to the base mesh which has been used in the previous static simulations. The mesh can only become finer than this, not coarser. The ``max refinement level`` is set at 3, giving a maximum possible number of cells of 3 million. However, the ``max number elements`` limits the number of cells to 50,000 to keep the simulation within feasible computational expense.
```

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+
+.. image:: images/adapt_without_mesh.png
+    :alt: velocity distribution 150
+    :align: center
+
+.. image:: images/adapt_with_mesh.png
+    :alt: pressure distribution 150
+    :align: center
+
+The resulting drag coefficient of 0.855 is more accurate than determined using the static mesh.
```
Comment (tonielgeitani) [inline doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:310]: How do you know it is more accurate? Do you have an analytical value for C_d for Re = 150? If so, you haven't mentioned it.

**Review by tonielgeitani — COMMENTED** [review]
Very minor comments :)

**Review by blaisb — COMMENTED** [review]
Some minor things
Very good work!

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst (hunk 6 above)
Comment (CaroleAnneDaunais) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:310]: The calculator site or an actual paper?

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+        subsection w
+            set Function expression = 0
+        end
+    end
+    subsection bc 2
+        set type              = slip
+    end
+  end
+
+There are three boundary conditions, as shown in the figure above. A ``noslip`` condition is applied on the surface of the sphere, where the velocity should be 0. The inlet velocity is set to `u=1m/s`, and the walls have a ``slip`` boundary condition.
```
[the remark this answers is not in the document]
Comment (CaroleAnneDaunais) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:102]: maybe parallel boundaries of the domain?

[the remark this answers is not in the document]
Comment (CaroleAnneDaunais) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:102]: or boundaries of the domain that are parallel to the flow direction

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst (hunk 4 above)
Comment (CaroleAnneDaunais) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:193]: Yeah I will remove that sentence altogether as it was previously mentioned. The other change it the viscosity but that was made clear in it's section

**Review by tonielgeitani — APPROVED** [review]

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst (hunk 6 above)
Comment (blaisb) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:310]: Use the calculator site. It agglomerates results from tons of paper anyway

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+    set subdivision             = 1
+  end
+
+The ``steady_bdf`` method solves for a steady-state simulation using adjoint time stepping with a bdf1 scheme. An initial time step is used to complete a transient iteration, and with each iteration, the time step is increased. The simulation is considered to have reached steady-state when the L2 norm of the initial residual is lower than stop tolerance at the start of a non-linear solution step, i.e. until the time step is large enough that a pseudo-steady-state has been reached.
+
+
+Initial condition
+~~~~~~~~~~~~~~~~~
+
+An initial condition is introduced for the Re = 150 problem. This way, a Re = 10 initial condition is set. Since the solution can easily be found at Re = 10, this is used as an initial attempt to hence find the solution at :math:`Re=150`.
```
Comment (Luckabarbeau) [inline doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:203]: suggestion : this is used as an initial guess to find the solution at :math:`Re=150`.

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+
+.. note::
+  The last line of the file shows the force calculated in the last iteration. Since the flow in the x-direction, the x-direction force ``f_x`` gives the drag force.
+
+.. code-block:: text
+
+  cells      f_x           f_y          f_z      
+  5823 98.3705224612 -0.0000000785 0.0000001119
+
+Given the flow parameters, the calculated drag coefficient is 250.50, using 6000 cells. At Re=0.1, an analytical solution of the drag coefficient is known: :math:`C_D = 240`. The deviation from the analytical solution is primarily due to the coarseness of the mesh. It would be relevant to carry out a mesh refinement analysis.
```
Comment (Luckabarbeau) [inline doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:282]: The deviation is much more likely to come from the size of the domain for a very low Reynolds number.

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+    :align: center
+.. image:: images/velocity_mesh_150.png
+    :alt: velocity distribution mesh 150
+    :align: center
+
+.. image:: images/pressure_150.png
+    :alt: pressure distribution 150
+    :align: center
+
+The drag coefficient at Re=150 using this example simulation is 0.798, against an analytical solution of 0.889.
```
Comment (Luckabarbeau) [inline doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:303]: There is no analytical solution for the drag at these Reynolds. ( yet) I would just talk say the name of the correlation that this value comes from.

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst [hunk 11]
```diff
@@ -0,0 +1,329 @@
...
+
+Third case results (Re = 150 with an adaptative mesh refinement)
+----------------------------------------------------------------
+
+Using mesh adaptative refinement, the final mesh contains slightly more than 101,000 cells. The resulting velocity profile is shown without and with the underlying mesh. Refinement around the sphere and wake can be observed:
```
Comment (Luckabarbeau) [inline doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:308]: Normal that you have 101k elements if you limited the number to 50k ?

**Review by Luckabarbeau — COMMENTED** [review]
Very nice Job !. Very small comments here and there

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst (hunk 10 above)
Comment (CaroleAnneDaunais) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:303]: Thank you for that!

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst (hunk 11 above)
Comment (CaroleAnneDaunais) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:308]: Ah

Comment (CaroleAnneDaunais) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:308]: @blaisb

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst (hunk 9 above)
Comment (CaroleAnneDaunais) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:282]: I will add this :)

File: doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst (hunk 11 above)
Comment (CaroleAnneDaunais) [reply doc/source/examples/incompressible-flow/3d-flow-around-sphere/flow-around-sphere.rst:308]: it is set to 100,000, which is a bit below 101k. I modified the parameters.

**Pull request merged**